File: ina_2sem/labor_cs/3_lab/1_task/Szesciokat.java
public class Szesciokat extends Figura {
    private final double bok;

    public Szesciokat(double bok) {
        if (bok <= 0) throw new IllegalArgumentException("Bok musi być dodatni.");
        this.bok = bok;
    }

    @Override
    public double obliczPole() {
        return (3 * Math.sqrt(3) * bok * bok) / 2;
    }

    @Override
    public double obliczObwod() {
        return 6 * bok;
    }

    @Override
    public String podajNazwe() {
        return "Sześciokąt foremny";
    }
}

File: ina_2sem/labor_cs/3_lab/2_task/Szesciokat.cpp
#include "Szesciokat.h"

Szesciokat::Szesciokat(double b) : bok(b) {
    if (b <= 0) throw std::invalid_argument("Bok musi być dodatni.");
}

double Szesciokat::obliczPole() {
    return (3 * sqrt(3) * bok * bok) / 2;
}

double Szesciokat::obliczObwod() {
    return 6 * bok;
}

std::string Szesciokat::podajNazwe() {
    return "Sześciokąt foremny";
}
